shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -429,17 +429,18 @@ trait TermImpl extends StatementImpl { self: Term =>
     case Tup(fields) => Tuple(fields.map(fld => (fld._1, fld._2 match {
       case Fld(m, s, v) => val ty = v.toType_!; Field(Option.when(m)(ty), ty)
     })))
-    case Bra(rcd, trm) if (!rcd) => trm.toType_!
+    case Bra(_, trm) => trm.toType_!
```

Comment:
I don't think that's a good idea. Why not match specifically on `{ <record> }` and `( <something> )`? I don't think `( <record> )` makes sense for example. Better be strict in what we accept to avoid surprises.

Reply:
Agree! But we now have `val bra = if (bk === Curly) Bra(true, Rcd(res.map { ...` in `NewParser.scala`. Isn't it enough? I'm not sure how to make it so far.

Reply:
No, it's not enough that this code exists in another file. The helper function in question could be used from other places than "NewParser".

Reply:
Is this way ok? 🤔

Reply:
Should be fine.